### PR TITLE
Add rewrite label attr expr visitor

### PR DESCRIPTION
**Pull request by yixinglu**
Refactor expression util implementation with expr visitor.

### CODE REVIEW

**Review by dutor — COMMENTED** [review]
IMO, here the __Find__ and __Collect__ visitor could be abstracted to two more high level kinds: FindAnyVistor and CollectAllVisitor, which you construct with a predicate.

BTW. there are several names better to be renamed, like:
 * findAnyKind -> findAny
 * findAllKindInAll -> collectAll
 * hasAnyKind -> hasAny

File: src/visitor/FindExprVisitor.h [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+    void visit(VertexExpression* expr) override;
+    void visit(EdgeExpression* expr) override;
+
+    void visitBinaryExpr(BinaryExpression* expr) override;
+
+    void findExpr(const Expression* expr);
+
+    bool found_{false};
+    const Expression* expr_{nullptr};
+    const std::unordered_set<Expression::Kind>& exprs_;
```
Comment (dutor) [inline src/visitor/FindExprVisitor.h:65]: `kinds_`

**Comment by yixinglu** [comment]
> IMO, here the __Find__ and __Collect__ visitor could be abstracted to two more high level kinds: FindAnyVistor and CollectAllVisitor, which you construct with a predicate.
> 
> BTW. there are several names better to be renamed, like:
>  * findAnyKind -> findAny
>  * findAllKindInAll -> collectAll
>  * hasAnyKind -> hasAny

good idea！i will update this pr according to your advice again.

File: src/visitor/FindAnyExprVisitor.cpp [hunk 2]
```diff
@@ -0,0 +1,149 @@
...
+    if (found_) return;
+    expr->operand()->accept(this);
+}
+
+void FindAnyExprVisitor::visit(FunctionCallExpression *expr) {
+    findExpr(expr);
+    if (found_) return;
+    for (const auto &arg : expr->args()->args()) {
+        arg->accept(this);
+        if (found_) break;
```
Comment (laura-ding) [inline src/visitor/FindAnyExprVisitor.cpp:34]: better to use `return`, keep the same as under code.

File: src/visitor/CollectAllExprsVisitor.h [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+#include "visitor/ExprVisitorImpl.h"
+
+namespace nebula {
+namespace graph {
+
+class CollectAllExprsVisitor final : public ExprVisitorImpl {
+public:
+    explicit CollectAllExprsVisitor(const std::unordered_set<Expression::Kind>& exprKinds);
+    bool ok() const override {
+        return true;
```
Comment (laura-ding) [inline src/visitor/CollectAllExprsVisitor.h:22]: return !exprs_.empty();

File: src/visitor/FindAnyExprVisitor.cpp (hunk 2 above)
Comment (yixinglu) [reply src/visitor/FindAnyExprVisitor.cpp:34]: fixed

File: src/visitor/CollectAllExprsVisitor.h (hunk 3 above)
Comment (yixinglu) [reply src/visitor/CollectAllExprsVisitor.h:22]: here should always return TRUE for traversing all nodes in expression tree.

File: src/visitor/FindExprVisitor.h (hunk 1 above)
Comment (yixinglu) [reply src/visitor/FindExprVisitor.h]: fixed

**Review by laura-ding — APPROVED** [review]

**Comment by yixinglu** [comment]
@dutor @laura-ding resolved conflicts. Please review this PR again.

**Review by laura-ding — APPROVED** [review]

**Review by dutor — APPROVED** [review]
Bravo!

**Pull request merged**